### PR TITLE
[L1] Move public headers to interface directory

### PR DESCRIPTION
**Pull request by smuzaffar**
As reported in #34718 , this PR moves headers in interface directory for `L1`

```
      2 src/L1Trigger/CSCTrackFinder/src/CSCTFDTReceiver.h
      1 src/L1Trigger/DTTrackFinder/src/L1MuDTTFConfig.h
      6 src/L1Trigger/TextToDigi/src/SourceCardRouting.h
```

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by rekovic** [comment]
+1

File: L1Trigger/TextToDigi/src/SourceCardRouting.cc [hunk 1]
```diff
@@ -4,7 +4,7 @@
 */
 
 // Prototype class definition
-#include "SourceCardRouting.h"  //hh"
+#include "../interface/SourceCardRouting.h"  //hh"
```
[the remark this answers is not in the document]
Comment (perrotta) [reply L1Trigger/TextToDigi/src/SourceCardRouting.cc:7]: @smuzaffar do you agree with this suggestion?

[the remark this answers is not in the document]
Comment (smuzaffar) [reply L1Trigger/TextToDigi/src/SourceCardRouting.cc:7]: [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1) allows the use of  `../interface/`  , so I would suggest to leave it as it is. If needed we can update it in an other PR

**Comment by perrotta** [comment]
+1

File: L1Trigger/TextToDigi/src/SourceCardRouting.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (perrotta) [reply L1Trigger/TextToDigi/src/SourceCardRouting.cc:7]: @smuzaffar I have merged, but in reality note 4 says:
```
Include only files that are in the current directory, e.g.
#include "some_header.h"
or in the .../interface directory, e.g
#include "Subsystem/Package/interface/some_header.h"
```
where for the `.../interface` directory it explicitely writes down the full path

[the remark this answers is not in the document]
Comment (smuzaffar) [reply L1Trigger/TextToDigi/src/SourceCardRouting.cc:7]: ah right, I will open a PR to fix this

[the remark this answers is not in the document]
Comment (slava77) [reply L1Trigger/TextToDigi/src/SourceCardRouting.cc:7]: perhaps we need to update the rules, it looks like `...` is getting confused to be a relative path, while it was meant to be an ellipsis

**Pull request merged**